### ISSUE TITLE
Method decorators in class expressions are silently ignored.

### ISSUE THREAD

**Issue by brandonbloom** [Microsoft/TypeScript#28903]
**TypeScript Version:**  3.2

**Search Terms:**

- decorators
- class expressions

**Code**

```ts
const decorator = () => null;

// This calls __decorate with decorator.
class Statement {
  @decorator()
  method() {

  }
}

// This doesn't! It also doesn't produce a compiler error of any kind.
(class Expression {
  @decorator()
  method() {

  }
})
```

**Expected behavior:**

__decorate should be called, or a compiler error should be produced.

**Actual behavior:**

The decorator is silently ignored, omitting it from the decoration compiled output.

**Playground Link:** 

https://www.typescriptlang.org/play/index.html#src=const%20decorator%20%3D%20()%20%3D%3E%20null%3B%0D%0A%0D%0Aclass%20Statement%20%7B%0D%0A%20%20%40decorator()%0D%0A%20%20method()%20%7B%0D%0A%0D%0A%20%20%7D%0D%0A%7D%0D%0A%0D%0A(class%20Expression%20%7B%0D%0A%20%20%40decorator()%0D%0A%20%20method()%20%7B%0D%0A%0D%0A%20%20%7D%0D%0A%7D)

**Related Issues:** 

- Using decorators on arguments in class expressions produces a compiler error: https://github.com/Microsoft/TypeScript/issues/7342

- This one is about decorators on class expressions, not methods within a class expression: https://github.com/Microsoft/TypeScript/issues/15975 - but produces some kind of error at least.

**Comment by weswigham** [comment]
At least in the playground, we issue a `Decorators are not valid here` error on the `@` on the decorator inside the class expression. What compiler options are you using where you don't see one?

**Comment by brandonbloom** [comment]
If you add `"experimentalDecorators": true` to your `tsconfig.json`, that error disappears.

**Comment by MartinJohns** [comment]
@brandonbloom Even with that flag the error remains:  
https://agentcooper.github.io/typescript-play/?experimentalDecorators=true#code/MYewdgzgLgBAJgU1AJwIZRMmBeGAKAShwD4YwBXAG0oG4AoO4S1CCGAZSnQQFsExYAbzowYAAUQp0mQiJh8oACxBxCMYXIC+dbXTxMWbAKIAPAA7IErAJbh1ciUkzTks0QuWqiG0ds0EGIA

**Comment by brandonbloom** [comment]
D'oh, figured it out. I'm testing with `allowJs: true` and the ignored decorator is in a `.js` file. This seems to me like it should be a syntax error or a code generation failure, not a type checking error.

**Comment by DanielRosenwasser** [comment]
So what's the deal, would we consider this a bug or a design limitation?

**Comment by RyanCavanaugh** [comment]
Hopefully in TS7 we can put grammar checks into their own out-of-band phase
